tests/test_sagemaker/test_sagemaker_processing.py

```diff
@@ -332,3 +357,144 @@ def test_list_processing_jobs_paginated_with_fragmented_targets():
         0
     )
     assert processing_jobs_with_2_next_next.get("NextToken").should.be.none
+
+
+@mock_sagemaker
+def test_add_tags_to_training_job(sagemaker_client):
+    processing_job_name = "MyProcessingJob"
+    role_arn = "arn:aws:iam::{}:role/FakeRole".format(ACCOUNT_ID)
+    container = "382416733822.dkr.ecr.us-east-1.amazonaws.com/linear-learner:1"
+    bucket = "my-bucket"
+    prefix = "my-prefix"
+    app_specification = {
+        "ImageUri": container,
+        "ContainerEntrypoint": ["python3", "app.py"],
+    }
+    processing_resources = {
+        "ClusterConfig": {
+            "InstanceCount": 2,
+            "InstanceType": "ml.m5.xlarge",
+            "VolumeSizeInGB": 20,
+        },
+    }
+    stopping_condition = {"MaxRuntimeInSeconds": 60 * 60}
+
+    job = MyProcessingJobModel(
+        processing_job_name,
+        role_arn,
+        container=container,
+        bucket=bucket,
+        prefix=prefix,
+        app_specification=app_specification,
+        processing_resources=processing_resources,
+        stopping_condition=stopping_condition,
+    )
+    resp = job.save(sagemaker_client)
+    resource_arn = resp["ProcessingJobArn"]
+
+    tags = [
+        {"Key": "myKey", "Value": "myValue"},
+    ]
+    response = sagemaker_client.add_tags(ResourceArn=resource_arn, Tags=tags)
+    assert response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    response = sagemaker_client.list_tags(ResourceArn=resource_arn)
+    assert response["Tags"] == tags
```

Comment:
To reduce duplicating the setup, I think we can test the deletion here as well.
```
massive setup
add tags
test tags exist
delete tags
test no tags exist (or only other tags are remaining..)
```
In general I'm in favor of having dedicated tests for separate functionality, but considering the size of the setup, and the fact that the core of the functionality is also covered in other tests, it's OK to test these features in a single method